Add docs for legacy URL aliases

jportner opened the pull request:
Resolves #110145.

I do not intend to backport this as it is not applicable to pre-8.0 versions of Kibana.

Docs preview links:

* [Legacy URL Aliases](https://kibana_110279.docs-preview.app.elstc.co/guide/en/kibana/master/legacy-url-aliases.html)
* [Disable legacy URL aliases API](https://kibana_110279.docs-preview.app.elstc.co/guide/en/kibana/master/spaces-api-disable-legacy-url-aliases.html)

jportner commented on docs/api/spaces-management.asciidoc:
```diff
@@ -20,10 +20,13 @@ The following {kib} spaces APIs are available:
 
 * <<spaces-api-resolve-copy-saved-objects-conflicts, Resolve copy saved objects to space conflicts API>> to overwrite saved objects returned as errors from the copy saved objects to space API
 
+* <<spaces-api-disable-legacy-url-aliases, Disable legacy URL aliases API>> to disable legacy URL aliases
```
I'll change it to "to disable legacy URL aliases if an error is encountered"

jportner commented on docs/developer/advanced/legacy-url-aliases.asciidoc:
```diff
@@ -0,0 +1,46 @@
+
+For example, consider the following scenario:
+
+You have {kib} 7.16 installed and you created a new dashboard. This dashboard's ID is "123". Then, you create a new space called "Bill's
```
In addition to this, I changed "you created a new dashboard" to "you create a new dashboard" :+1:

jportner commented on docs/developer/advanced/legacy-url-aliases.asciidoc:
```diff
@@ -0,0 +1,46 @@
+
+You use these two dashboards frequently, so you use your web browser to add bookmarks for them. After some time, you decide upgrade to {kib}
+8.0. When these two dashboards go through the conversion process, the one in "Bill's space" will have its ID changed to "456". Now, the URL
+to access that dashboard is different -- not to worry though, there is a legacy URL alias for that dashboard!
```
I can't help it if I'm excited about this 😄

jportner commented on docs/developer/advanced/legacy-url-aliases.asciidoc:
```diff
@@ -0,0 +1,45 @@
+in this situation, you might want to disable the legacy URL alias completely. This leaves the saved object intact and you will not lose any
+data by disabling the alias -- you just won't be able to use the old URL to access that saved object anymore.
+
+To disable a legacy URL alias, you'll need three pieces of information about it: the `targetSpace`, the `targetType`, and the `sourceId`.
```
@gchaps based on your comments below, it sounds like you think this is clear enough with your suggested changes?

gchaps commented on docs/developer/advanced/legacy-url-aliases.asciidoc:
```diff
@@ -0,0 +1,45 @@
+
+Legacy URL aliases are intended to be fully transparent, but there are some very rare situations where this can lead to an error. For
+example, you may have a dashboard and one of the visualizations may fail to load, directing you to this page. If you encounter an error
+in this situation, you might want to disable the legacy URL alias completely. This leaves the saved object intact and you will not lose any
```
I think the text is good as it is, especially since the only way to get into the situation is by using third-party integrations.  I don't feel a screenshot is necessary.

gchaps commented on docs/api/spaces-management/disable_legacy_url_aliases.asciidoc:
```diff
@@ -0,0 +1,59 @@
+  (Required, object array) The aliases to disable.
++
+.Properties of `aliases`
+[%collapsible%open]
```
I wouldn't spend too much time on this.  The formatting also works find as it is.

jportner replied:
I tried one more thing and it's fixed now 😄 
Needed to add `[role="child_attributes"]` before the section.

gchaps commented on docs/developer/advanced/legacy-url-aliases.asciidoc:
```diff
@@ -0,0 +1,45 @@
+* *Default space*: `http://localhost:5601/app/dashboards#/view/123`
+* *Bill's space*: `http://localhost:5601/s/bills-space/app/dashboards#/view/123`
+
+You use these two dashboards frequently, so you bookmark them in your web browser. After some time, you decide upgrade to {kib} 8.0. When
```
```suggestion
You use these two dashboards frequently, so you bookmark them in your web browser. After some time, you decide to upgrade to {kib} 8.0. When
```

jportner replied:
Good catch!

gchaps reviewed (approved):
LGTM with one minor edit.

jportner commented:
> I do not intend to backport this as it is not applicable to pre-8.0 versions of Kibana.

I updated the UI components in #114172 and backported it to 7.x, and I added doc links to these docs in that PR. So I need to backport this too.
